### PR TITLE
AC: Added adapter for Ultra-Lightweight Face Detection models

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 1]
```diff
@@ -909,3 +909,84 @@ def softmax(x):
...
+        return parameters
+
+    def configure(self):
+        self.scores_out = self.get_value_from_config('scores_out')
+        self.boxes_out = self.get_value_from_config('boxes_out')
+
+        # Set default values
+        self.prob_threshold = 0.7
+        self.iou_threshold = 0.5
+
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:943]: Possibly it makes sense to make these parameters configurable

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 2]
```diff
@@ -909,3 +909,84 @@ def softmax(x):
...
+                mask = score > self.prob_threshold
+                filtered_boxes, filtered_score = boxes[mask, :], score[mask]
+                if filtered_score.size == 0:
+                    continue
+
+                x_mins = (filtered_boxes[:, 0])
+                y_mins = (filtered_boxes[:, 1])
+                x_maxs = (filtered_boxes[:, 2])
+                y_maxs = (filtered_boxes[:, 3])
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:966]: Why do you wrap output to the brackets?
```suggestion
                x_mins, y_mins, x_maxs, y_maxs = filtered_boxes.T
```

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 3]
```diff
@@ -909,3 +909,84 @@ def softmax(x):
...
+        raw_outputs = self._extract_predictions(raw, frame_meta)
+
+        batch_scores = raw_outputs[self.scores_out]
+        batch_boxes = raw_outputs[self.boxes_out]
+
+        result = []
+        for identifier, scores, boxes in zip(identifiers, batch_scores, batch_boxes):
+            detections = {'labels': [], 'scores': [], 'x_mins': [], 'y_mins': [], 'x_maxs': [], 'y_maxs': []}
+
+            for label, score in enumerate(np.transpose(scores)):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:954]: there is only 2 labels, possibly can just get class-1 instad of iteration and skipping label-0?

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 4]
```diff
@@ -909,3 +909,84 @@ def softmax(x):
...
+                filtered_boxes, filtered_score = boxes[mask, :], score[mask]
+                if filtered_score.size == 0:
+                    continue
+
+                x_mins = (filtered_boxes[:, 0])
+                y_mins = (filtered_boxes[:, 1])
+                x_maxs = (filtered_boxes[:, 2])
+                y_maxs = (filtered_boxes[:, 3])
+
+                keep = NMS.nms(x_mins, y_mins, x_maxs, y_maxs, filtered_score, self.iou_threshold,
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:968]: is it impossible to make NMS in postprocessing (not in adapter), as well as scores filtering?

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 3 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py:954]: or generalize adapter not only for face detection

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py:943]: Done

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 2 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py]: Done

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 3 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py]: Done

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 4 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py]: Moved to postprocessing

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 5]
```diff
@@ -909,3 +909,67 @@ def softmax(x):
...
+            if filtered_score.size != 0:
+                x_mins, y_mins, x_maxs, y_maxs = filtered_boxes.T
+
+                labels = np.full_like(filtered_score, 1, dtype=int)
+                detections['labels'].extend(labels)
+                detections['scores'].extend(filtered_score)
+                detections['x_mins'].extend(x_mins)
+                detections['y_mins'].extend(y_mins)
+                detections['x_maxs'].extend(x_maxs)
+                detections['y_maxs'].extend(y_maxs)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:966]: why do you need this, you can simplify to return just arrays without storing them in dictionary

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py [hunk 6]
```diff
@@ -909,3 +909,52 @@ def softmax(x):
...
+
+        batch_scores = raw_outputs[self.scores_out]
+        batch_boxes = raw_outputs[self.boxes_out]
+
+        result = []
+        for identifier, scores, boxes in zip(identifiers, batch_scores, batch_boxes):
+            score = np.transpose(scores)[1]
+            mask = score > self.score_threshold
+            filtered_boxes, filtered_score = boxes[mask, :], score[mask]
+            x_mins, y_mins, x_maxs, y_maxs = filtered_boxes.T
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection.py:955]: but you still have situation when model does not have boxes filtered by threshold, I think you need handle this case

Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py]: Done

File: tools/accuracy_checker/accuracy_checker/adapters/detection.py (hunk 5 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection.py]: Done

**Review by eaidova — APPROVED** [review]

**Pull request merged**